tools/test/test_depsolve.py

```diff
@@ -1409,29 +1522,135 @@ def test_depsolve(tmp_path, repo_servers, dnf_config, detect_fn, with_sbom, test
                 assert res["solver"] == "dnf"
 
 
+@pytest.mark.parametrize("with_sbom", [False, True])
+@pytest.mark.parametrize("dnf_config, detect_fn", [
+    ({}, assert_dnf),
+    ({"use_dnf5": False}, assert_dnf),
+    ({"use_dnf5": True}, assert_dnf5),
+], ids=["no-config", "dnf4", "dnf5"])
+def test_depsolve_sbom(repo_servers, dnf_config, detect_fn, with_sbom):
+    try:
+        detect_fn()
+    except RuntimeError as e:
+        pytest.skip(str(e))
+
+    test_case = depsolve_test_case_basic_2pkgs_2repos
+    transactions = test_case["transactions"]
+    tc_repo_servers = get_test_case_repo_servers(test_case, repo_servers)
+    repo_configs = [gen_repo_config(server) for server in tc_repo_servers]
+
+    with TemporaryDirectory() as cache_dir:
+        res, exit_code = depsolve(transactions, cache_dir, dnf_config, repo_configs, with_sbom=with_sbom)
+
+        if test_case.get("error", False):
```

Comment:
AIUI this test runs the scenario `depsolve_test_case_basic_2pkgs_2repos` which is a single test case where we expect success. But the test is written in a generic way to also pass failure scenarios. Is the plan to add more scenarios to this test? Or am I missing something?

Or maybe a shared helper to check the depsolve result (but I noticed that the checks are all slightly different or maybe that is not possible)

Reply:
The depsolve result check is always a bit different, but extracting the bare minimum into a common function is probably doable. The case of checking for errors is more of a copy & paste leftover. :innocent: